### DISCUSSION TITLE
Is there a way to show what has been synced in Web UI?

### DISCUSSION THREAD

**Discussion by dkmcgowan** [philosowaffle/peloton-to-garmin#511] [General]
I use to use the headless container but stopped once I enabled 2FA because of ECG on my venu, got curious and clicked it and then no way to remove it...

I came back and see 2FA works now and a new web ui!  I set everything up and it's working, I can go to manual sync and did a ride today and it showed up on the home page (actually really fast, like within minutes, which is confusing considering I left the default check every 24 hours).  I was able to sync all of July and click the home page button and sync today, yay!

My question is am I missing something seeing how you can see what has been synced and what hasn't been?  Like if I did another ride today and it showed up on the home page, they both look the same.

Same for the list on the sync tab, I'm not sure how many I did the sync for last night when testing things.

In the headless there use to be a file that maintained what all had already been successful pushed to Garmin

**Comment by philosowaffle** [comment]
Hi, welcome back!

No there is no way to see what has already been synced via the UI.  After I realized that Garmin does a sufficient job at rejecting duplicate uploads, I decided to employ the KISS principle and not have P2G attempt to keep track of uploads anymore.

It should be safe to re-sync workouts from P2G without fear of duplicates on the Garmin side.

> it showed up on the home page (actually really fast, like within minutes, which is confusing considering I left the default check every 24 hours).

Yup, the UI makes on-demand requests to Peloton, and while it does implement some caching, you can usually expect that it will be fetching the latest data.  The scheduled sync behavior is purely for background syncing in the event someone doesn't want to visit the UI every time they need to sync a workout.  Though note that background sync does not support 2FA accounts.
